src/main/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -267,15 +264,15 @@ else if (!(entity instanceof MobEntity))
      * @param maximumLevel the maximum level for the tool to find.
      * @return true if tool is acceptable
      */
-    public static boolean hasToolLevel(@Nullable final ItemStack stack, final IToolType toolType, final int minimalLevel, final int maximumLevel)
+    public static boolean hasToolLevel(@Nullable final ItemStack stack, final ToolTypeEntry toolType, final int minimalLevel, final int maximumLevel)
     {
         if (isEmpty(stack))
         {
             return false;
         }
 
-        final int level = Compatibility.isTinkersWeapon(stack) ? Compatibility.getToolLevel(stack) : getMiningLevel(stack, toolType);
-        return isTool(stack, toolType) && verifyToolLevel(stack, level, minimalLevel, maximumLevel);
+        final int level = Compatibility.isTinkersWeapon(stack) ? Compatibility.getToolLevel(stack) : toolType.getMiningLevel(stack);
+        return toolType.checkIsTool(stack) && verifyToolLevel(stack, level, minimalLevel, maximumLevel);
```

Comment:
I asked this before, but could this method be reduced to
```java
return toolType.checkIsTool(stack) && verifyToolLevel(stack, toolType.getMiningLevel(stack), minimalLevel, maximumLevel);
```
Leave the compatibility stuff to the tool type logic if applicable.